pkg/operators/spoke/controller.go

```diff
@@ -2,28 +2,395 @@ package spoke
+			_, err := n.nucleusClient.Update(ctx, spokeCore, metav1.UpdateOptions{})
+			return err
+		}
+	}
+
+	// SpokeCore is deleting, we remove its related resources on spoke
+	if !spokeCore.DeletionTimestamp.IsZero() {
+		if err := n.cleanUp(ctx, controllerContext, config); err != nil {
+			return err
+		}
+		return n.removeWorkFinalizer(ctx, spokeCore)
+	}
+
+	// Start deploy spoke core components
+	// Check if namespace exists
+	_, err = n.kubeClient.CoreV1().Namespaces().Get(ctx, config.SpokeCoreNamespace, metav1.GetOptions{})
+	if err != nil {
+		helpers.SetNucleusCondition(&spokeCore.Status.Conditions, nucleusapiv1.StatusCondition{
+			Type:    spokeCoreApplied,
+			Status:  metav1.ConditionFalse,
+			Reason:  "SpokeCoreApplyFailed",
+			Message: fmt.Sprintf("Failed to get namespace %q", config.SpokeCoreNamespace),
+		})
+		helpers.UpdateNucleusSpokeStatus(
+			ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(spokeCore.Status.Conditions...))
+		return err
+	}
+
+	// Check if bootstrap secret exists
+	_, err = n.kubeClient.CoreV1().Secrets(config.SpokeCoreNamespace).Get(
+		ctx, config.BootStrapKubeConfigSecret, metav1.GetOptions{})
+	if err != nil {
+		helpers.SetNucleusCondition(&spokeCore.Status.Conditions, nucleusapiv1.StatusCondition{
+			Type:    spokeCoreApplied,
+			Status:  metav1.ConditionFalse,
+			Reason:  "SpokeCoreApplyFailed",
+			Message: fmt.Sprintf("Failed to get bootstracp secret %s/%s", config.SpokeCoreNamespace, config.BootStrapKubeConfigSecret),
+		})
+		helpers.UpdateNucleusSpokeStatus(
+			ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(spokeCore.Status.Conditions...))
+		return err
+	}
+
+	// Deploy the static resources
+	// Apply static files
+	resourceResults := resourceapply.ApplyDirectly(
+		resourceapply.NewKubeClientHolder(n.kubeClient),
+		controllerContext.Recorder(),
+		func(name string) ([]byte, error) {
+			return assets.MustCreateAssetFromTemplate(name, bindata.MustAsset(filepath.Join("", name)), config).Data, nil
+		},
+		"manifests/spoke/spoke-clusterrolebinding.yaml",
+		"manifests/spoke/spoke-serviceaccount.yaml",
+	)
+	errs := []error{}
+	for _, result := range resourceResults {
+		if result.Error != nil {
+			errs = append(errs, fmt.Errorf("%q (%T): %v", result.File, result.Type, result.Error))
+		}
+	}
+
+	if len(errs) > 0 {
+		appleErros := operatorhelpers.NewMultiLineAggregate(errs)
+		helpers.SetNucleusCondition(&spokeCore.Status.Conditions, nucleusapiv1.StatusCondition{
+			Type:    spokeCoreApplied,
+			Status:  metav1.ConditionFalse,
+			Reason:  "SpokeCoreApplyFailed",
+			Message: appleErros.Error(),
+		})
+		helpers.UpdateNucleusSpokeStatus(
+			ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(spokeCore.Status.Conditions...))
+		return appleErros
+	}
+
+	// Create hub config secret
+	hubSecret, err := n.kubeClient.CoreV1().Secrets(config.SpokeCoreNamespace).Get(
```

Comment:
I don't think I see how you can create this secret.  Where can you get it from?  If the user provides it to you, couldn't the first cut here be the user writing it into the target namespace directly?

Reply:
I think the bootstrap secret is provided by user, this secret is updated by the registration agent to fill the kubeconfig with cert/key.